charts/tfy-karpenter-config/templates/gpu/karpenter-gpu-default-ec2nodeclass.yaml

```diff
@@ -58,15 +58,29 @@ spec:
         encrypted: {{ .Values.karpenter.gpuDefaultNodeTemplate.encrypted }}
   {{- if not (eq .Values.karpenter.gpuDefaultNodeTemplate.userData "") }}
   {{- if eq .Values.karpenter.gpuDefaultNodeTemplate.userData "default" }}
+  {{- if and .Values.karpenter.gpuDefaultNodeTemplate.enableSoci (not (eq (include "gpuDefaultNodeTemplate.inferredAmiFamily" .) "bottlerocket")) }}
```

Comment:
nit, this is getting harder to read, I would prefer nesting with simpler conditions


```
if inferredAmiFamily "al2"

if enableSoci
...
else
...

else if inferredAmiFamily "al2023"

if enableSoci
...
else
...

else if inferredAmiFamily "bottlerocket"
...

```